src/en/index.md

```diff
@@ -627,6 +629,134 @@ func fetchAll() async {
+
+```swift
+guard let first = await group.next() else {
+    group.cancelAll()
+    return 0
+}
+// first == 1 or first == 2
+let second = await group.next() ?? 0
+group.cancelAll()
+// second == 2 or second == 1
+return first + second
+```
+
+You can learn more about [TaskGroup](https://developer.apple.com/documentation/swift/taskgroup) in the Swift documentation.
+
+#### Note about Tasks and SwiftUI.
+
+When writing a UI, you often want to start asynchronous tasks from a synchronous context. For example, you want to asynchronously load an image as a response to a UI element touch. Starting asynchronous tasks from a synchronous context is not possible in Swift. This is why you see solutions involving `Task { ... }`, which introduces unmanaged tasks.
+
+You can't use `TaskGroup` from a synchronous SwiftUI modifier because `withTaskGroup()` is an async function too and so are its related functions.
+
+As an alternative, SwiftUI offers an asynchronous modifier that you can use to start asynchronous operations. The `.task { }` modifier, which we already mentioned, accepts a `() async -> Void` function, ideal for calling other `async` functions. It is available on every `View`. It is triggered before the view appears and the tasks it creates are managed and bound to the lifecycle of the view, meaning the tasks are cancelled when the view disappears.
+
+Back to the tap-to-load-an-image example: instead of invoking an asynchronous `loadImage()` function from a synchronous `.onTap() { ... }` function, you can create an asynchronous stream of events that will be handled by `async` functions.
+
+Here is a simple example:
+
+```swift
+struct ContentView: View {
+    
+    // the list of events the UI can generate
+    enum Event {
+        case buttonClicked
+    }
+    
+    var stream: AsyncStream<Event>? = nil
+    var continuation: AsyncStream<Event>.Continuation? = nil
+    
+    init() {
+        self.stream = AsyncStream { continuation in
+            self.continuation = continuation
+        }
+    }
+    var body: some View {
+        Button("Click Me !") {
+            // produce an event
+            self.continuation?.yield(.buttonClicked)
+        }
+        // the View manages the subtask
+        // it starts before the view is displayed 
+        // and stops when the view is hidden
+        .task {
+            // we are in an async context, we can call an async function
+            await processEventStream()
+        }
+    }
+    
+    // This is an async function
+    func processEventStream() async throws {
+        guard let stream = self.stream else { return }
+
+        // consume UI events
+        for try await event in stream {
+            
+            switch event {
+            // asynchronously load the image when the button is clicked  
+            case .buttonClicked:
+              try await loadImage()
+            }
+        }
+    }
+}
+```
```

Comment:
The `AsyncStream` pattern here is quite elaborate. A simpler alternative using `.task(id:)` could achieve the same result:

```swift
struct ContentView: View {
    @State private var shouldLoadImage = false
    
    var body: some View {
        Button("Load Image") {
            shouldLoadImage = true
        }
        .task(id: shouldLoadImage) {
            guard shouldLoadImage else { return }
            await loadImage()
        }
    }
}
```

This pattern reruns the task when `id` changes - much easier to understand for readers new to Swift concurrency.